### ISSUE TITLE
Support mixed references to roles and collections when defining requirements

### ISSUE THREAD

**Issue by morningspace** [crossplane-contrib/provider-ansible#81]
### What problem are you facing?

It looks in the current implementation, when defining roles and collections as requirements, they are mutually exclusive, while in Ansible requirements.yml, it supports mixed of both.

### How could Crossplane help solve your problem?

We can support it in next releases, so that aligns with what Ansible requirements.yml natively support. For example:

```yaml
apiVersion: ansible.crossplane.io/v1alpha1
kind: ProviderConfig
metadata:
  name: provider-config-example
spec:
  requirements: |
    ---
    roles:
      # Install a role from Ansible Galaxy.
      - name: sample_namespace.sample_role
        version: 0.1.0
    collections:
      # Install a collection from Ansible Galaxy.
      - name: sample_namespace.sample_collection
        version: 0.1.0
        source: https://galaxy.ansible.com
```

**Comment by ron1** [comment]
[Here:](https://github.com/tektoncd/catalog/blob/65c6285ffb8e7f2350c12f872ab085d970fef4b0/task/ansible-runner/0.1/ansible-runner.yaml#L53-L59), I see Tekton supports installing both roles and collections by simply invoking ansible-galaxy twice, once for each. Is the expectation that provider-ansible will do likewise?

**Comment by fahedouch** [comment]
> [Here](https://github.com/tektoncd/catalog/blob/65c6285ffb8e7f2350c12f872ab085d970fef4b0/task/ansible-runner/0.1/ansible-runner.yaml#L53-L59) I see Tekton supports installing both roles and collections by simply invoking ansible-galaxy twice, once for each. Is the expectation that provider-ansible will do likewise?

Not really, we support installing roles and collection by single invoke of `ansible-galaxy`. To make it happens, The installation requirement is forwarded by the `requirement.yml` file which combine role and collection. Then we delegate to the `ansible-galaxy` the responsibility to execute the install

**Comment by fahedouch** [comment]
> > [Here](https://github.com/tektoncd/catalog/blob/65c6285ffb8e7f2350c12f872ab085d970fef4b0/task/ansible-runner/0.1/ansible-runner.yaml#L53-L59) I see Tekton supports installing both roles and collections by simply invoking ansible-galaxy twice, once for each. Is the expectation that provider-ansible will do likewise?
> 
> Not really, we have to support installing roles and collection by single invoke of `ansible-galaxy`. To make it happens, The installation requirement is forwarded by the `requirement.yml` file which combine role and collection. Then we delegate to the `ansible-galaxy` the responsibility (of `how to do it` ) to execute the install

Looks like `ansible-galaxy` cannot handle both roles and requirement installation from a `single cmd`.a  twice call STGM (cc @ron1 )

**Comment by fahedouch** [comment]
> @fahedouch Now that issue #173 is resolved, it seems like an easy fix for this issue would be to change
> 
> https://github.com/crossplane-contrib/provider-ansible/blob/6094036ac684e319332713c4854d92ac12afdfad/internal/controller/ansibleRun/ansibleRun.go#L255-L281
> 
> so that the single switch on `requirementsType` is changed to 2 conditionals, one for `role` and one for `collection`. Is this low-enough hanging fruit to get this in the 0.4.1 release?

@ron1  I think So, it might be included in the `v0.4.1` . I am planning to release the `v0.4.1` this night or tomorrow.
